app/services/container_dashboard_service.rb

```diff
@@ -78,4 +80,71 @@ def providers
       }
     ]
   end
+
+  def heatmaps
+    # Get latest daily rollup for each node.
+    metrics = MetricRollup.where(:resource_type => ContainerNode.name, :capture_interval_name => "daily")
+    metrics = metrics.where(:resource_id => @ems.container_nodes) if @ems
+    metrics = metrics.order(:resource_id, :timestamp => :desc)
+    metrics = metrics.select('DISTINCT ON (metric_rollups.resource_id) metric_rollups.*')
+    metrics = metrics.includes(:resource)
+
+    {
+      :nodeCpuUsage    => metrics.collect { |m|
+        avg_cpu = m.cpu_usage_rate_average.round(2)
+        {
+          :id      => m.resource_id,
+          :tooltip => "Name: #{m.resource.name} - #{avg_cpu} percent of total",
+          :value   => avg_cpu / 100.0 # 1% should be 0.01
+        }
+      },
+      :nodeMemoryUsage => metrics.collect { |m|
+        avg_mem = m.mem_usage_absolute_average.round(2)
+        {
+          :id      => m.resource_id,
+          :tooltip => "Name: #{m.resource.name} - #{avg_mem} percent of total",
+          :value   => avg_mem / 100.0 # 1% should be 0.01
+        }
+      }
+    }
+  end
+
+  def node_utilization
+    metrics = MetricRollup.where(:resource_type => ContainerNode.name, :capture_interval_name => "daily")
+    metrics = metrics.where(:resource_id => @ems.container_nodes) if @ems
+    metrics = metrics.select("sum(derived_vm_numvcpus * cpu_usage_rate_average / 100) as used_cores,
+                              sum(mem_usage_absolute_average * derived_memory_available / 100) as used_mem,
+                              sum(derived_vm_numvcpus) as total_cores,
+                              sum(derived_memory_available) as total_mem,
+                              cast(timestamp as date) as day")
+    metrics = metrics.order("day").group("day")
```

Comment:
It feels wrong, but are you doing it because the metrics don't exist?  It the metrics don't exist, they should be added to the metrics tables as possibly derived columns and then rolled up.

What concerns me most here is that the rollup calculations are aware of TimeProfiles, and this isn't.